exporter/opensearchexporter/go.mod

```diff
@@ -5,6 +5,7 @@ go 1.22.0
 require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden v0.114.0
 	github.com/opensearch-project/opensearch-go/v2 v2.3.0
+	github.com/opensearch-project/opensearch-go/v4 v4.3.0
```

Comment:
Why does this seem to be adding `v4` rather than replacing `v2`?

Reply:
this change will require the update of the code that uses `github.com/opensearch-project/opensearch-go/v2` to `github.com/opensearch-project/opensearch-go/v4` and a go mod tidy as well to clean up the references to `v2`